File: Coding-Interviews/32-I-从上到下打印二叉树.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
#include <queue>
#include <vector>
using namespace std;
class Solution
{
public:
    vector<int> levelOrder(TreeNode* root)
    {
        if (!root)
            return {};
        vector<int>      ret;
        queue<TreeNode*> q;
        q.push(root);
        while (!q.empty()) {
            auto cnt = q.size();
            for (int i = 0; i < cnt; i++) {
                auto top = q.front();
                q.pop();
                ret.push_back(top->val);
                if (top->left)
                    q.push(top->left);
                if (top->right)
                    q.push(top->right);
            }
        }
        return ret;
    }
};

File: Coding-Interviews/32-I-从上到下打印二叉树.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
import queue


class Solution:
    def levelOrder(self, root: TreeNode) -> List[int]:
        if not root:
            return []
        res = []
        q = queue.Queue()
        q.put(root)
        while not q.empty():
            node = q.get()
            res.append(node.val)
            if node.left:
                q.put(node.left)
            if node.right:
                q.put(node.right)
        return res
